app/src/main/java/com/io/movio/fragments/MovieDetailFragment.kt

```diff
@@ -0,0 +1,59 @@
+package com.io.movio.fragments
+
+import android.os.Bundle
+import android.text.method.ScrollingMovementMethod
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.io.movio.databinding.FragmentMovieDetailBinding
+import com.io.movio.models.Movie
+
+// the fragment initialization parameters
+private const val ARG_MOVIE = "movie"
+
+
+class MovieDetailFragment : Fragment() {
+    private lateinit var binding: FragmentMovieDetailBinding
+    private lateinit var movie: Movie
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            movie = it.getSerializable(ARG_MOVIE) as Movie
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+            binding.tvTitle.text = movie.title
+            binding.tvCast.text = movie.cast
+            binding.tvDescription.text = movie.description
+            binding.tvDescription.movementMethod = ScrollingMovementMethod()
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?): View {
+        binding = FragmentMovieDetailBinding.inflate(inflater , container , false)
+
+        return binding.root
+    }
+
+    companion object {
+        /**
+         * Use this factory method to create a new instance of
+         * this fragment using the provided parameters.
+         */
+        @JvmStatic
+        fun newInstance(movie: Movie) =
+            MovieDetailFragment().apply {
+                arguments = Bundle().apply {
+                    putSerializable("movie" , movie)
```

Comment:
Serializable is okay, the better aproach is Parcelable, you can read about the differences between both and if you want you can try to use it.